File: problem-026/solution.js
function reciprocalCycles(n) {
    let longestCycle = 0;
    let answer = 0;

    for (let d = 2; d < n; d++) {

        let remainders = [];
        let remainder = 1;
        let position = 0;

        while (remainder !== 0 && remainders[remainder] === undefined) {

            remainders[remainder] = position;

            remainder = (remainder * 10) % d;

            position++;
        }

        if (remainder !== 0) {
            let cycleLength = position - remainders[remainder];

            if (cycleLength > longestCycle) {
                longestCycle = cycleLength;
                answer = d;
            }
        }
    }

    return answer;
}

File: problem-026/solution.py
def reciprocal_cycles(n):
    longest_cycle = 0
    answer = 0

    for d in range(2, n):

        remainders = {}
        remainder = 1
        position = 0

        while remainder != 0 and remainder not in remainders:

            remainders[remainder] = position

            remainder = (remainder * 10) % d

            position += 1

        if remainder != 0:
            cycle_length = position - remainders[remainder]

            if cycle_length > longest_cycle:
                longest_cycle = cycle_length
                answer = d

    return answer
